service/history/workflow/task_generator.go

```diff
@@ -54,7 +53,7 @@ type (
 		) error
 		GenerateDeleteExecutionTask(
 			now time.Time,
-		) error
+		) (*tasks.DeleteExecutionTask, error)
```

Comment:
return `tasks.Task` or `[]tasks.Task` please.

Once queue refactor change is done, I am going to update all the generator methods to return the generated task, instead of adding it to ms directly.

Reply:
"Depend on interface, return concrete type", right? Unless there is a good reason for exception.

Reply:
Since we control the caller code and we can prevent them from being closely coupled, I don't really have a strong opinion here.